azurerm/internal/services/recoveryservices/resource_arm_backup_policy_vm.go

```diff
@@ -319,12 +325,18 @@ func resourceArmBackupProtectionPolicyVMCreateUpdate(d *schema.ResourceData, met
 		return fmt.Errorf("The Azure API has recently changed behaviour so that provisioning a `count` for the `retention_daily` field can no longer be less than 7 days for new/updates to existing Backup Policies. Please ensure that `count` is less than 7, currently %d", d.Get("retention_daily.0.count").(int))
 	}
 
+	var instantRestoreDays *int32
+	if instantRestoreDaysRaw := d.Get("instant_restore_retention_days"); instantRestoreDaysRaw != nil {
+		instantRestoreDays = utils.Int32(int32(instantRestoreDaysRaw.(int)))
+	}
+
 	policy := backup.ProtectionPolicyResource{
 		Tags: tags.Expand(t),
 		Properties: &backup.AzureIaaSVMProtectionPolicy{
-			TimeZone:             utils.String(d.Get("timezone").(string)),
-			BackupManagementType: backup.BackupManagementTypeAzureIaasVM,
-			SchedulePolicy:       expandArmBackupProtectionPolicyVMSchedule(d, times),
+			TimeZone:                      utils.String(d.Get("timezone").(string)),
+			BackupManagementType:          backup.BackupManagementTypeAzureIaasVM,
+			SchedulePolicy:                expandArmBackupProtectionPolicyVMSchedule(d, times),
+			InstantRpRetentionRangeInDays: instantRestoreDays,
```

Comment:
From the tests it appears that this needs to be absent if `nil`, or the API returns:
`Error: Error creating/updating Azure Backup Protection Policy "acctest-201015105922143565" (Resource Group "acctestRG-backup-201015105922143565"): backup.ProtectionPoliciesClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="UserErrorInvalidRequestParameter" Message="Parameter InstantRpRetentionRangeInDays in request is invalid\r\nPlease provide correct value for parameter InstantRpRetentionRangeInDays"`

Reply:
I've gone and done this - the tests appear to have passed.